src/server/req_runjob.c

```diff
@@ -484,66 +478,36 @@ req_runjob(struct batch_request *preq)
 	/* OK - go back over the run job request, assign the vhosts */
 	/* and finally run the job by calling req_runjob2()	    */
 
-	if (jt == IS_ARRAY_NO) {
+	if (jt == IS_ARRAY_Single) {
 
-		/* just a regular job, pass it on down the line and be done */
-		if (call_to_process_hooks(preq, hook_msg, sizeof(hook_msg),
-				pbs_python_set_interrupt) == 0) {
-			reply_text(preq, PBSE_HOOKERROR, hook_msg);
-			return;
-		}
-		pjob = where_to_runjob(preq, parent);
-		if (pjob) {
-			/* free prov_vnode before use */
-			job_attr_def[(int)JOB_ATR_prov_vnode].at_free(
-				&pjob->ji_wattr[(int)JOB_ATR_prov_vnode]);
-			req_runjob2(preq, parent);
-		}
-		return;
+	/* single subjob, if parent qeueud, it can be run */
```

Comment:
I see the above comment here. But where are we making sure of the following.
if  parent is not queued then the request should be rejected(since it might be already running)

Reply:
This comment copied from previous code line 506. As per the existing arrayjob design the subjob is issued to req_runjob() by scheduler only when parent is queued.